#166237757 initialize express server with environment variables

obayomi96 opened the pull request:
#### What does this PR do?
This PR Initializes the express server with environment variables

#### Description of Task to be completed?
Dev script to start the project in development mode (using nodemon)
start script to start the project in production mode (using node)
Integrate dotenv for managing environment variables
ignore .env in .gitginore
Readme.md file updated with instructions on how to run the project
The WIldcard route to handle any endpoint, return a JSON message 'hello books deferral'

Have the following endpoints working
localhost:4000/api/v1

#### How should this be manually tested?
> Clone the Hello-Books repository 
>$ git clone https://github.com/los-deferal-02/Hello-Books.git
>$ cd Hello-Books
>$ npm install
>$ npm run Start-dev
> Navigate to the endpoint stated above

#### What are the relevant pivotal tracker stories?
#166237757


![Screenshot (234)](https://user-images.githubusercontent.com/43539944/58431086-1f527400-80a4-11e9-985a-e45d2b80b5cb.png)

obayomi96 committed:
chore(Express server):initialize
Dev script to start the project in development mode (uding nodemon)
start script to start the project in production mode (using node)
Integrate dotenv for managing environment variables
ignore .env in .gitginore
Readme.md file updated with instructions on how to run the project
WIldcard route to handle any endpoint, return a json message 'hello books deferral' [delivers #166237757]

obayomi96 committed:
chore(Express server):initialize
Dev script to start the project in development mode (uding nodemon)
start script to start the project in production mode (using node)
Integrate dotenv for managing environment variables
ignore .env in .gitginore
Readme.md file updated with instructions on how to run the project
WIldcard route to handle any endpoint, return a json message 'hello books deferral' [delivers #166237757]

ayodejiAA commented:
Looks okay. Nice work.

chialuka commented:
Looks good. Thank you for handling the README too

tolulope-od reviewed (commented):
This is great stuff @obayomi96 I noticed to added a catch-all for all undefined routes. I think it wouldn't be a bad idea to add an error handling middleware also. Do you think you can add that to this branch? Again, good stuff. Thanks for kicking things off for us 👍

obayomi96 commented:
> This is great stuff @obayomi96 I noticed to added a catch-all for all undefined routes. I think it wouldn't be a bad idea to add an error handling middleware also. Do you think you can add that to this branch? Again, good stuff. Thanks for kicking things off for us 👍

Thanks @tolulope-od  It's a good idea to implement that as it didn't occur to me before.  I will have to do it at night. If other PRs have been merged I will do it in another branch as a chore. Thanks for pointing that out

ayodejiAA commented:
I went through it again and found most babel packages were installed as dependencies. Can you move them to dev, leaving only babel polyfill which is required only in production?

obayomi96 commented:
> I went through it again and found most babel packages were installed as dependencies. Can you move them to dev, leaving only babel polyfill which is required only in production?

True @Xwebyna, I did that initially, and it really gave me a taugh time. Untill I had to leave it in the dependency. The app kept breaking. Maybe if someones moves it we'll see

ayodejiAA commented:
> > I went through it again and found most babel packages were installed as dependencies. Can you move them to dev, leaving only babel polyfill which is required only in production?
> 
> True @Xwebyna, I did that initially, and it really gave me a taugh time. Untill I had to leave it in the dependency. The app kept breaking. Maybe if someones moves it we'll see

That's okay.  @obayomi96. One of us will look into it. Great work.

akpante3 commented on index.js:
```diff
@@ -0,0 +1,25 @@
+    console.log(`Server started on port ${port}`);
+});
+
+export default app;
```
@obayomi96 nice work here, please create a new line here.

obayomi96 replied:
Noted, will do that now.

akpante3 commented on routes/index.js:
```diff
@@ -0,0 +1,11 @@
+    });
+});
+
+export default router;
```
@obayomi96 new line here too

obayomi96 replied:
Noted

obayomi96 committed:
Fix conflits

obayomi96 committed:
Fix conflits

obayomi96 committed:
Fix empty line

chialuka committed:
chore(eslint): configure eslint for the project
- setup ESLint with Airbnb style guide
- add .babelrc file
- correct errors flagged by ESLint
[Finishes #166237752]

chialuka committed:
chore(eslint): configure eslint for the project
- setup ESLint with Airbnb style guide
- add .babelrc file
- correct errors flagged by ESLint
[Finishes #166237752]

chialuka committed:
Merge branch 'ch-configure-eslint-166237752' of
github.com:los-deferal-02/Hello-Books
into ch-configure-eslint-166237752

chialuka committed:
Merge branch 'ch-configure-eslint-166237752' of
github.com:los-deferal-02/Hello-Books
into ch-configure-eslint-166237752

chialuka committed:
Merge branch 'ch-configure-eslint-166237752' of github.com:los-deferal-02/Hello-Books into ch-configure-eslint-166237752

chialuka committed:
add .hound.yml file

akpante3 committed:
Merge pull request #3 from los-deferal-02/ch-configure-eslint-166237752

#166237752- Configure ESLint with AirBnB style Guide
